version: "3"
services:
  pyglidein-server:
    image: wipac/pyglidein_server:1.0
    networks:
      app_net:
    hostname: pyglidein-server
    volumes:
      - "../:/pyglidein"
      - "~/.pyglidein_secrets_server:/home/condor/.pyglidein_secrets"
    ports:
      - "8001:8001"
  pyglidein-client:
    image: wipac/pyglidein_client_centos${OS_VERSION:-7}:1.0
    networks:
      app_net:
    hostname: pyglidein-client
    privileged: true
    volumes:
      - "/etc/OpenCL:/etc/OpenCL"
      - "../:/pyglidein"
      - "~/.pyglidein_secrets:/home/pyglidein/.pyglidein_secrets"
      - nvidia_driver_384.90:/usr/local/nvidia:ro
    devices:
      - /dev/nvidia0:/dev/nvidia0
      - /dev/nvidiactl:/dev/nvidiactl
      - /dev/nvidia-uvm:/dev/nvidia-uvm
  graphite-dev:
   image: graphiteapp/graphite-statsd
   networks:
     app_net:
   hostname: graphite-dev
   ports:
    - "80:80"
    - "2003-2004:2003-2004"
    - "2023-2024:2023-2024"
    - "8125:8125/udp"
    - "8126:8126"
networks:
  app_net:
    driver: bridge
    ipam:
      driver: default
      config:
        -
          subnet: 172.22.0.0/24
volumes:
  nvidia_driver_384.90:
    external: true
